### PR TITLE
Add CpuType definitions for Genoa

### PR DESCRIPTION
**Pull request by secretfader**

### CODE REVIEW

File: src/guest/measure/vcpu_types.rs [hunk 1]
```diff
@@ -59,6 +68,9 @@ impl CpuType {
             CpuType::EpycMilan => cpu_sig(25, 1, 1),
             CpuType::EpycMilanV1 => cpu_sig(25, 1, 1),
             CpuType::EpycMilanV2 => cpu_sig(25, 1, 1),
+            CpuType::EpycGenoa => cpu_sig(25, 17, 1),
+            CpuType::EpycGenoaV1 => cpu_sig(25, 17, 1),
+            CpuType::EpycGenoaV2 => cpu_sig(25, 17, 1),
```
Comment (hwrdtm) [inline src/guest/measure/vcpu_types.rs:73]: where did you get these magical parameters from?

**Review by hwrdtm — APPROVED** [review]

File: src/guest/measure/vcpu_types.rs (hunk 1 above)
Comment (glitch003) [reply src/guest/measure/vcpu_types.rs:73]: yeah maybe worth dropping a comment in the code about where they came from.  could be a link to AMD docs.

**Comment by secretfader** [comment]
I pulled them from the CPU information in Linux, cross referenced with AMD
docs. Notes should be in the commit messages.

File: src/guest/measure/vcpu_types.rs [hunk 2]
```diff
@@ -17,6 +17,9 @@ pub enum CpuType {
     EpycMilan = 10,
     EpycMilanV1 = 11,
     EpycMilanV2 = 12,
+    EpycGenoa = 13,
```
Comment (Garandor) [inline src/guest/measure/vcpu_types.rs:20]: I'm confused, if each of the series has its own CPU Type, why are we hardcoding it to EPYCv4 in https://github.com/LIT-Protocol/lit-assets/blob/703e35dd5e6dd20b8a3d5fc831391331c54f3188/rust/lit-os/lit-cli-os/src/cmd/os/guest/instance/create.rs#L582
which seems to not even be appropriate for Milan

Comment (secretfader) [reply src/guest/measure/vcpu_types.rs:20]: @Garandor Ideally, we remove some of the deeply nested dependencies in a future refactor, but I figure it is best to define these in the root crate, which is imported [here](https://github.com/LIT-Protocol/lit-assets/blob/ef0f4294389df705941707f329ab6a72f799e2e9/rust/lit-os/lit-os-core/src/guest/types.rs#L5). 

I assume I'll need to remove the hard-coded conversions there as well, but we use those types in the build manifest, and the `from_str` conversion from external types relies on this implementation as far as I'm aware.

Comment (Garandor) [reply src/guest/measure/vcpu_types.rs:20]: All for removing the `GuestCPUType` if we can do so safely (only issue coming to mind is if we were to `migrate` a Guest VM from a Milan host to a Genoa host, but migration is unsupported anyway)

i'm not familiar with the build mf code, just as a heads up: GuestCpuType also hardcodes the `from_str` to `EPYCv4`

https://github.com/LIT-Protocol/lit-assets/blob/703e35dd5e6dd20b8a3d5fc831391331c54f3188/rust/lit-os/lit-os-core/src/guest/types.rs#L137

**Pull request merged**